### PR TITLE
cleanup: simplify Bazel targets for new libraries

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8225)

### CODE REVIEW

File: BUILD.bazel [hunk 1]
```diff
@@ -20,117 +20,126 @@ exports_files([
...
+]
+
+[cc_library(
+    name = "experimental-{library}".format(library = library),
+    deps = [
+        "//google/cloud/{library}:google_cloud_cpp_{library}".format(library = library),
+    ],
+) for library in EXPERIMENTAL_LIBRARIES]
+
+[cc_library(
```
Comment (dbolduc) [inline BUILD.bazel:105]: optionally, you can combine these into the same for loop

File: BUILD.bazel [hunk 2]
```diff
@@ -20,117 +20,126 @@ exports_files([
...
+    name = "experimental-{library}".format(library = library),
+    deprecation = "this library is now GA, please use //:{library} instead.".format(library = library),
+    tags = ["manual"],
     deps = [
-        "//google/cloud/assuredworkloads:google_cloud_cpp_assuredworkloads",
+        "//google/cloud/{library}:google_cloud_cpp_{library}".format(library = library),
     ],
-)
+) for library in TRANSITION_LIBRARIES]
```
Comment (dbolduc) [inline BUILD.bazel:119]: same option

File: BUILD.bazel [hunk 3]
```diff
@@ -20,117 +20,126 @@ exports_files([
...
+    "workflows",
+]
+
+TRANSITION_LIBRARIES = [
+    "secretmanager",
+    "tasks",
+]
+
+GA_LIBRARIES = [
+]
```
Comment (dbolduc) [inline BUILD.bazel:96]: do you want to add `bigquery`, `iam`, `pubsub`, `spanner` to here?

(leaving their deprecated targets below)

**Review by dbolduc — APPROVED** [review]
just some options that you might have already considered and decided against

File: BUILD.bazel (hunk 1 above)
Comment (coryan) [reply BUILD.bazel:105]: I do not think you can, these are list comprehensions, if you write `[x(a) y(a) for a in list]` that is a syntax error, if you write `[(x(a), y(a)) for a in list]` that is a list of tuples, and you don't want tuples...

File: BUILD.bazel (hunk 3 above)
Comment (coryan) [reply BUILD.bazel:96]: I am leaving those for a later refactoring.

File: BUILD.bazel (hunk 2 above)
Comment (coryan) [reply BUILD.bazel:119]: Ditto, but if you know of a Python / Starlark trick to get these working I would be glad to know.

**Pull request merged**